instance/management/commands/migrate_swift_to_s3.py

```diff
@@ -0,0 +1,135 @@
+import subprocess
+import logging
+
+from django.core.management.base import BaseCommand
+
+from instance.models.openedx_instance import OpenEdXInstance
+import swiftclient
+
+LOG = logging.getLogger(__name__)
+
+# Classes #####################################################################
+
+
+class Command(BaseCommand):
+    """
+    Migrate SWIFT to S3.
+    """
+    help = (
+        'Migrate all servers\' data from SWIFT to S3.'
+    )
+
+    def __init__(self, *args, **kwargs):
+        super(Command, self).__init__(*args, **kwargs)
+        self.options = {}
+        self.retried = {}
+
+    def _get_swift_connection(self, instance):
+        return swiftclient.Connection(
+            user=instance.swift_openstack_user,
+            key=instance.swift_openstack_password,
+            authurl=instance.swift_openstack_auth_url,
+            tenant_name=instance.swift_openstack_tenant,
+            auth_version='2',
+            os_options={'region_name': instance.swift_openstack_region}
+        )
+
+    def _create_rclone_config(self, access_key, secret_key):
+        """
+        Creates rclone config, to ease the process of copying
+        """
+        rclone_config = "rclone config create ocim-s3 s3 access_key_id %s secret_access_key %s" % (access_key, secret_key)
+        print("Would run this command: %s.       ← FIXME test it and run it" % rclone_config)
+        return
+
+        subprocess.Popen(rclone_config % (access_key, secret_key), shell=True)
+
+
+    def _copy_subdirectory(self, key_name, new_name, swift_container, s3_bucket):
+        """
+        Copy subdirectory using rclone
+        """
+        LOG.info("%s ---> %s" % (key_name, new_name))
+        command = "rclone copy ocim-swift:%s/%s ocim-s3:%s/%s" % (swift_container, key_name, s3_bucket, new_name)
+        print("Would run this command: %s.       ← FIXME test it and run it" % command)
+        return
+        subprocess.Popen(command, shell=True)
+
+    def _migrate_swift_to_s3(self, instance):
+        # FIXME Will these 2 values already be stored in the instance? If not, where to get them from? (Or should we set them and save?)
+        # FIXME and in particular: will the S3 region match the SWIFT region? (some clients don't want their data in a particular region; see comments on ticket)
+        self._create_rclone_config(instance.s3_access_key, instance.s3_secret_access_key)
+
+        copied = []  # FIXME describe variable
+        swift = self._get_swift_connection(instance)
+
+        container = instance.swift_container_name
+        s3_bucket = instance.s3_bucket_name  # FIXME will this instance already have the right bucket name? If not, we must set it and save the instance
+        LOG.info("Migrating container: %s" % container)
+        for item in swift.get_container(container)[1]:
+            LOG.info("Seeing file: %s", item['name'])
+
+            # FIXME test this part; will it also copy files in the root? (in addition to directories):
```

Comment:
Agree that this is something to test, I think it will copy them, and if it does I don't think it's needed to filter by directory as you are doing below